src/Libraries/CoreNodeModels/Input/ColorPalette.cs

```diff
@@ -27,25 +31,65 @@ namespace CoreNodeModels.Input
     [OutPortDescriptions("Selected Color.")]
     public class ColorPalette : NodeModel
     {
-        private DSColor dscolor = DSColor.ByARGB(255,0,0,0);
-        public DSColor dsColor
+        private DSColor dscolor = DSColor.ByARGB(255, 0, 0, 0);
+        private DSColor prevColor = DSColor.ByARGB(255, 0, 0, 0);
+        private bool Isundo = false;
```

Comment:
In what case do you expect this `Isundo` to be used? I only see it used in the setter. Also the naming is a bit confusing, can we make it more clearer by renaming to `HasUndoableChange` or `IsUndoable`, or something else